modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/HeavyOrLightWeightAppControlTestCase.java

```diff
@@ -0,0 +1,201 @@
+  private static final String APP_XML_FILE = "simple.xml";
+  private static final String HEAVYWEIGHT_APP = "heavyweight";
+  private static final String LIGHTWEIGHT_APP = "lightweight";
+  private static final String MULE_CONNECTORS_GROUP_ID = "org.mule.connectors";
+  private static final String MULE_PLUGIN_CLASSIFIER = "mule-plugin";
+  private static final String MULESOFT_PUBLIC_REPOSITORY = "https://repository.mulesoft.org/nexus/content/repositories/public/";
+  private static final TestLogger logger = getTestLogger(DefaultArchiveDeployer.class);
+
+  @Parameterized.Parameters(name = "Parallel: {0}")
+  public static List<Boolean> params() {
+    return asList(false);
+  }
+
+  public HeavyOrLightWeightAppControlTestCase(boolean parallelDeployment) {
+    super(parallelDeployment);
+  }
+
+  @Test
+  @Story(LIGHTWEIGHT)
+  public void lightweightAppDeploymentDownloadsDependenciesFromRemoteRepo() throws Exception {
+    final File muleRepository = Paths.get(getMuleBaseFolder().getAbsolutePath(), "repository").toFile();
+
+    final MavenConfigurationBuilder mavenConfigurationBuilder = newMavenConfigurationBuilder();
+    mavenConfigurationBuilder.remoteRepository(newRemoteRepositoryBuilder()
+        .id("mulesoft-public")
+        .url(new URL(MULESOFT_PUBLIC_REPOSITORY))
+        .build());
+    mavenConfigurationBuilder.localMavenRepositoryLocation(muleRepository);
+    setMavenConfig(mavenConfigurationBuilder.build());
+
+    final ArtifactPluginFileBuilder mulePlugin = new ArtifactPluginFileBuilder("mule-sockets-connector")
+        .withGroupId(MULE_CONNECTORS_GROUP_ID)
+        .withVersion("1.2.0")
+        .withClassifier(MULE_PLUGIN_CLASSIFIER);
+
+    final ApplicationFileBuilder applicationFileBuilder = appFileBuilder(LIGHTWEIGHT_APP)
+        .definedBy(APP_XML_FILE)
+        .dependingOn(mulePlugin)
+        .usingLightWeightPackage();
+
+    addPackedAppFromBuilder(applicationFileBuilder);
+    startDeployment();
+
+    final String applicationName = applicationFileBuilder.getDeployedPath();
+    final File applicationRepository = Paths.get(getAppFolder(applicationName).toString(), "repository").toFile();
+    final Collection<File> muleRepositoryContents = getRepositoryContents(muleRepository);
+    final List<String> muleRepositoryContentNames = muleRepositoryContents.stream().map(File::getName).collect(toList());
+
+    assertDeploymentSuccess(applicationDeploymentListener, applicationFileBuilder.getId());
+    assertApplicationAnchorFileExists(applicationName);
+    assertThat(applicationRepository.exists(), is(false));
+    assertThat(muleRepositoryContentNames, hasItem(allOf(startsWith("mule-sockets-connector"), endsWith("-mule-plugin.jar"))));
+    assertThat(muleRepositoryContentNames, not(hasItem(allOf(startsWith("mule-core"), endsWith(".jar")))));
+  }
+
+  @Test
+  @Story(LIGHTWEIGHT)
+  public void lightweightAppDeploymentFailsIfNoRemoteRepoConfigured() throws Exception {
```

Comment:
add @Description

Reply:
If I add a description, I think I would be repeating what the name of the test says. What do you think?

Reply:
The @description is used for the documentation. Probably after the refactor this has to reevaluated.